design-proposals/project-deploymnet.md

```diff
@@ -0,0 +1,188 @@
+
+After completing all the prerequisites, the site administrator can edit and apply a ClusterLink CRD instance to the operator. The operator will then create the ClusterLink components, including:
+
+* A ClusterLink dedicated namespace, which by default will be cl-operator-ns.
+* Deployment of ClusterLink controlplane components, including controlplane-pod, controlplane-service, and RBAC roles.
+* Deployment of ClusterLink dataplane components, including dataplane pods (single or multiple) and the dataplane-service.
+* Deployment of ClusterLink ingress, providing an external access point using load-balancer/node-port/gateway-API service.
+
+Overall, the ClusterLink deployment stages are:
+
+<img src="deployment.png" width="800" height="400" alt="ClusterLink Deployment Stages"/>
+
+The ClusterLink operator will have privileged permissions, allowing it to create a dedicated namespace and the ClusterLink components within.
+The ClusterLink components will be created in the `clusterlink-system-ns``, and the control-plane will have privileged permissions within the namespace (for creating K8s services) and watch permissions in other namespaces.
+Additionally, the operator will create and update components in response to the CRD instance create or update actions. Once created or updated, the operator will not actively monitor the state of each component, assuming that any changes will be made only by privileged users.
+Furthermore, the operator will delete all components and the namespace of ClusterLink when the CRD instance is deleted.
+The ClusterLink deployment is limited to one instance per cluster. In the case of deploying more than one CRD instance, only the first one will be taken into account, and the others will be ignored.
+
+The deployment file for the operator and example of CRD instance will be included in every project release.
+
+The ClusterLink CLI will be utilized to automate the deployment process.
+For instance, the following commands can assist the fabric administrator in creating the site certificates:
+
+    clusterlink create fabric --name <fabric_name>
+    clusterlink create site --name <site_name> --fabric <fabric_name> 
+
+For the site administrator, automation of deploying the certificates as a secret to the cluster, deploying the ClusterLink operator, and the CRD instance can be achieved with:
+
+    clusterlink install --site <site_name>
+
+Note: The ClusterLink CLI uses the kubeconfig, and it should be set to the specific cluster that needs to be configured
+
+### ClusterLink CRD
+
+The ClusterLink CRD includes the following fields:
+
+* **API version:** clusterlink.net/v1alpha1
+* **Kind:** clusterlink
+
+* **Spec:**
+  
+    | Field name | Subfield name| Description | Default value |
+    | ---- | ----- | ------ | ----|
+    |dataplane | | ClusterLink dataplane component attributes||
+    |  |type| Types of dataplane, supported values "go" or "envoy"|"envoy"|
+    | |replicas| Number of dataplane replicas|1|
```

Comment:
Since there's no secret object reference, you may want to add to the prerequisites that the site certificate must be added  as a predefined name. You should also discuss the namespace (operator or clusterlink) since volume mounts work in the same namespace only.